patches/minecraft/net/minecraft/world/inventory/GrindstoneMenu.java.patch

```diff
@@ -1,5 +1,45 @@
 --- a/net/minecraft/world/inventory/GrindstoneMenu.java
 +++ b/net/minecraft/world/inventory/GrindstoneMenu.java
+@@ -35,6 +_,7 @@
+       }
+    };
+    private final ContainerLevelAccess f_39561_;
++   private int xp = -1;
+ 
+    public GrindstoneMenu(int p_39563_, Inventory p_39564_) {
+       this(p_39563_, p_39564_, ContainerLevelAccess.f_39287_);
+@@ -45,12 +_,12 @@
+       this.f_39561_ = p_39568_;
+       this.m_38897_(new Slot(this.f_39560_, 0, 49, 19) {
+          public boolean m_5857_(ItemStack p_39607_) {
+-            return p_39607_.m_41763_() || p_39607_.m_150930_(Items.f_42690_) || p_39607_.m_41793_();
++            return true;
```

Comment:
This may cause a desync between a forge server and a vanilla client or vice versa, have you tested that case? Should be able to launch both through the dev environment, use offline mode for the server.

Reply:
I've checked forge server vanilla client, the items would go in the grindstone no problem. I've tried to join a vanilla world from lan using forge, but that did not work (I couldn't join). I'll need to setup a vanilla server to test this?

Reply:
If you remove the --uuid 0 arg from the forge run config then you should be able to join LAN stuff

Reply:
Thanks for the help. I don't see any obvious issues. When forge is the server any item can go in the slot no issue. If vanilla is the server the item won't go in (if you press really fast it will go in for a fraction of a second, but it's hard to notice).